File: app/src/main/java/com/irishinterest/irishinterest/network/api/irishInterest/observer/GuiToModel.java
package com.irishinterest.irishinterest.network.api.irishInterest.observer;

import java.util.Observer;

public interface GuiToModel {
  void publishMessageModel(GuiMessage guiMessage);
  void registerObserver(Module module, Observer o);
  void unregisterObserver(Module module, Observer o);
}

File: app/src/main/java/com/irishinterest/irishinterest/network/api/irishInterest/observer/GuiToModel.kt
package com.irishinterest.irishinterest.network.api.irishInterest.observer

import java.util.*

interface GuiToModel {
    fun publishMessageModel(guiMessage: GuiMessage<*>)
    fun registerObserver(module: Module, o: Observer)
    fun unregisterObserver(module: Module, o: Observer)
}